packages/notifi-dapp-example/src/components/TopicStackRow.tsx

```diff
@@ -1,34 +1,73 @@
 import { Icon } from '@/assets/Icon';
+import { useUserInputParmToFilterOption } from '@/hooks/useUserInputParmToFilterOption';
 import {
+  TopicStackAlert,
   getFusionEventMetadata,
+  isEqual,
   useNotifiTopicContext,
 } from '@notifi-network/notifi-react';
 import React from 'react';
 
-import { TopicStandaloneRowMetadata } from './TopicList';
+import { TopicRowCategory } from './TopicList';
+import { TopicRowProps, isTopicGroupRow } from './TopicRow';
 import { TopicStack } from './TopicStack';
 import { TopicStackRowInput } from './TopicStackRowInput';
 
-export type TopicStackRowProps = TopicStandaloneRowMetadata;
-
-export const TopicStackRow: React.FC<TopicStackRowProps> = (props) => {
+export const TopicStackRow = <T extends TopicRowCategory>(
+  props: TopicRowProps<T>,
+) => {
+  const isTopicGroup = isTopicGroupRow(props);
   const { getTopicStackAlerts } = useNotifiTopicContext();
-  if (!props.topic.fusionEventDescriptor.id) return null;
+  const benchmarkTopic = isTopicGroup ? props.topics[0] : props.topic;
+  const fusionEventTypeId = benchmarkTopic.fusionEventDescriptor.id;
+  if (!fusionEventTypeId) return null;
 
-  const topicStackAlerts = getTopicStackAlerts(
-    props.topic.fusionEventDescriptor.id,
-  );
+  const topicStackAlerts = getTopicStackAlerts(fusionEventTypeId);
   const customIconUrl =
-    getFusionEventMetadata(props.topic)?.uiConfigOverride?.customIconUrl ?? '';
+    getFusionEventMetadata(benchmarkTopic)?.uiConfigOverride?.customIconUrl ??
+    '';
   const icon =
-    getFusionEventMetadata(props.topic)?.uiConfigOverride?.icon ?? 'INFO';
+    getFusionEventMetadata(benchmarkTopic)?.uiConfigOverride?.icon ?? 'INFO';
   const title =
-    getFusionEventMetadata(props.topic)?.uiConfigOverride?.topicDisplayName ??
-    '';
-
+    getFusionEventMetadata(benchmarkTopic)?.uiConfigOverride
+      ?.topicDisplayName ?? '';
+  const isTradingPairAlert = title.includes('Pair');
   const [isTopicStackRowInputVisible, setIsTopicStackRowInputVisible] =
     React.useState(topicStackAlerts.length > 0 ? false : true);
 
+  const alertsStack = React.useMemo(() => {
```

Comment:
We have filter out duplicate alerts with same filterOptions and made sure we have the duplicate alerts only render once here @eric-notifi